### ISSUE TITLE
Incorrect display of OpenGL 3 shadows

### ISSUE THREAD

**Issue by piramidamsc** [godotengine/godot#88123]
### Tested versions

Building the engine v4.3 dev3 win64

### System information

win10 ,opengl3,GTX 560 ti

### Issue description

shadows are displayed incorrectly in opengl 3 mode, the problem occurs when importing a 3d model in glb format, there are no problems when using obj format.
![S](https://github.com/godotengine/godot/assets/48522040/ae993cd6-c483-421d-91be-17e5edfd0e64)
![r](https://github.com/godotengine/godot/assets/48522040/be9d5dfd-1370-426d-aac1-7938295ee07c)

### Steps to reproduce

Create a new project with the opengl 3 visualization system, and export the 3d model in the format.glb without a second UV channel, then generate a uv2 light map in the editor.
[Новый игровой проект.zip](https://github.com/godotengine/godot/files/14216171/default.zip)

### Minimal reproduction project (MRP)

N / A

**Comment by Calinou** [comment]
This is likely due to mesh LOD being used too early. You can see the issue going away if you disable **Generate > LODs** in the Import dock for the `.glb` file.

**Comment by piramidamsc** [comment]
> * Related to [Regression in mesh LOD behavior with shadows since 4.0.beta3 #67479](https://github.com/godotengine/godot/issues/67479).
> 
> This is likely due to mesh LOD being used too early. You can probably see the issue going away if you disable **Generate > LODs** in the Import dock for the `.glb` file.

this is not related to the Lords generator.The error occurs if create shadow meshes is active

**Comment by Calinou** [comment]
The expected appearance is the one from CSG meshes and OBJ (notice how their shadows connect to the ground properly).

**Comment by piramidamsc** [comment]
> Can you clarify what is the expected correct look in 4.3-dev2?
> 
> I tested on my system (Linux, AMD) and the MRP looks the same in 4.3-dev2 and 4.3-dev3, as well as 4.2-stable.


I just double-checked and indeed starting from version 4.2 beta 1 to 4.3 dev 3 the problem persists, but disabling the creation of shadow meshes in the import menu solves the problem.

**Comment by piramidamsc** [comment]
Apparently shadow meshes are distorted when they are generated.

**Comment by piramidamsc** [comment]
> Can you clarify what is the expected correct look in 4.3-dev2?
> 
> I tested on my system (Linux, AMD) and the MRP looks the same in 4.3-dev2 and 4.3-dev3, as well as 4.2-stable.

Yes, this is my mistake, I just checked the builds from version 4.2 beta 1 to 4.3 dev 3 creating a new project for each version and the error popped up in all cases.

**Comment by clayjohn** [comment]
Taking a look at this locally. It seems the issue is not present when using the imported meshes directly even if they have UV2s generated. 

The only meshes that are impacted are ones where Arraymeshes are being used directly (i.e. not connected to a GLB or OBJ file). So, the clear workaround for now is to just reimport your meshes.

Seeing that you are using dev builds, I wonder if the ArrayMeshes were extracted during a period where mesh compatibility was broken. For example, if you had extracted those meshes during the Dev phase of 4.2, they could have been broken in a way that is fixed in newer builds. But you have preserved the broken mesh data by extracting them

**Comment by piramidamsc** [comment]
> Taking a look at this locally. It seems the issue is not present when using the imported meshes directly even if they have UV2s generated.
> 
> The only meshes that are impacted are ones where Arraymeshes are being used directly (i.e. not connected to a GLB or OBJ file). So, the clear workaround for now is to just reimport your meshes.
> 
> Seeing that you are using dev builds, I wonder if the ArrayMeshes were extracted during a period where mesh compatibility was broken. For example, if you had extracted those meshes during the Dev phase of 4.2, they could have been broken in a way that is fixed in newer builds. But you have preserved the broken mesh data by extracting them

Today I tested shadow mesh with a mesh format.gdb and they didn't have any problems with the grid and the transform, but apparently when I apply them to the model, their size and position are distorted. I don't have time to dig into the source code, so I leave this entertainment activity to you.

I will post the project for review.

[new_project_shadow_mesh.zip](https://github.com/godotengine/godot/files/14453047/new_project_shadow_mesh.zip)

**Comment by clayjohn** [comment]
I'm not sure if there is anything actionable anymore as this issue can't really be reproduced and seems to just be an issue with those specific ArrayMeshes. 

@piramidamsc Can you confirm that the issue went away after reimporting your GLTF meshes?

**Comment by Calinou** [comment]
Closing due to lack of response. Please comment if you can still reproduce this following the instructions in https://github.com/godotengine/godot/issues/88123#issuecomment-2093935607.

**Comment by piramidamsc** [comment]
> I'm not sure if there is anything actionable anymore as this issue can't really be reproduced and seems to just be an issue with those specific ArrayMeshes.
> 
> @piramidamsc Can you confirm that the issue went away after reimporting your GLTF meshes?

Hello.Sorry for taking so long to answer.I was busy.

Yes, I still have a problem.when generating UV2 for the lighting map, my shadows break.Maybe it's related to my PC since my hardware is outdated, gtx 560 ti and intel core i5 760.

I have prepared an archive for you with a .blend file from which I exported the model into an empty Godot 4.2.2 project, after which I clicked on the button to generate a scan for the entire maps.

[poy_test.zip](https://github.com/user-attachments/files/15920958/poy_test.zip)
![Screenshot_1](https://github.com/godotengine/godot/assets/48522040/88560396-5689-4d65-a719-3caafd20c101)
![Screenshot_2](https://github.com/godotengine/godot/assets/48522040/2ee57864-96bb-4f83-ad3d-4aa11aa9d979)
![Screenshot_3](https://github.com/godotengine/godot/assets/48522040/db675497-e076-40ee-8d40-76b983bc04c6)
